lib/Dialect/TritonNvidiaGPU/Transforms/TensorMemoryAllocation.cpp

```diff
@@ -334,12 +334,6 @@ class TritonTensorMemoryAllocationPass
     // if totalMemorySize > 512 we exceeded the maximum amount of tensor memory,
     // let the compilation finish so that we can raise an exception in python
     // for auto-tuner.
-    if (totalMemorySize > 0) {
-      assert(mod->getAttr("ttg.shared") != nullptr &&
-             cast<IntegerAttr>(mod->getAttr("ttg.shared")).getInt() != 0 &&
-             "Shared memory is required for allocation of Tensor Core memory.");
-    }
```

Comment:
I think this assertion is just wrong? Alright, if you're using tmem in a realistic scenario you likely have a tcgen05 mma that reads something from shared, but it prevents testing load and store in isolation.

Reply:
it is needed because the lowering of tmem alloc will implicitly use the top of smem without explicitly allocating.

Reply:
we could probably fix it a different way by making sure there is a minimum amount of smem allocated whenever tmem is not zero, I guess it never happened so far

Reply:
PTAL, updated the code to ensure at least 4 bytes of shared memory is allocated instead of failing.